plugin/auth_socket/auth_socket.c

```diff
@@ -119,13 +119,20 @@ static int socket_auth(MYSQL_PLUGIN_VIO *vio, MYSQL_SERVER_AUTH_INFO *info)
   u = cred.uid;
 #endif
 
-  /* and find the username for this uid */
+  /* and find the socket user name for this uid */
   getpwuid_r(u, &pwd_buf, buf, sizeof(buf), &pwd);
   if (pwd == NULL)
     return CR_ERROR;
 
-  /* now it's simple as that */
-  return strcmp(pwd->pw_name, info->user_name) ? CR_ERROR : CR_OK;
+  /* fill in the external user name used */
+  strncpy(info->external_user, pwd->pw_name, sizeof(info->external_user) - 1);
+  info->external_user[sizeof(info->external_user) - 1]= 0;
```

Comment:
In https://github.com/MariaDB/server/commit/567b68129943a1cceab1d7b4c68e2a4ba011cdc0, @Chaloff added the `safe_strcpy` function which _always_ NUL-terminates the destination string.

This could be used here. Simpler, shorter, less error-prone…

```
safe_strcpy(info->external_user, sizeof(info->external_user), pwd->pw_name);
```

Reply:
Good point. Updated in commit to use `safe_strcpy`.